src/test/java/com/meilisearch/integration/SettingsTest.java

```diff
@@ -476,50 +508,129 @@ public void testUpdateDistinctAttributeSettings() throws Exception {
         String initialDistinctAttribute = index.getDistinctAttributeSettings();
         String newDistinctAttribute = "title";
 
         index.waitForTask(index.updateDistinctAttributeSettings(newDistinctAttribute).getUid());
         String updatedDistinctAttribute = index.getDistinctAttributeSettings();
 
-        assertEquals(newDistinctAttribute, updatedDistinctAttribute);
-        assertNotEquals(initialDistinctAttribute, updatedDistinctAttribute);
-
         index.waitForTask(index.resetDistinctAttributeSettings().getUid());
         String distinctAttributeAfterReset = index.getDistinctAttributeSettings();
 
+        assertEquals(newDistinctAttribute, updatedDistinctAttribute);
+        assertNotEquals(initialDistinctAttribute, updatedDistinctAttribute);
+
         assertNotEquals(updatedDistinctAttribute, distinctAttributeAfterReset);
         assertNotEquals(initialDistinctAttribute, updatedDistinctAttribute);
     }
 
+    /** Tests of the typo tolerance setting methods */
     @Test
-    @DisplayName("Test reset ranking rules when null value is passed")
-    public void testUpdateRankingRuleSettingsUsingNull() throws Exception {
-        Index index = createIndex("testUpdateRankingRuleSettingsUsingNull");
-        String[] initialRankingRule = index.getRankingRuleSettings();
-        String[] newRankingRules = {
-            "typo",
-            "words",
-            "sort",
-            "proximity",
-            "attribute",
-            "exactness",
-            "release_date:desc",
-            "rank:desc"
-        };
+    @DisplayName("Test get typo tolerance settings by uid")
+    public void testGetTypoTolerance() throws Exception {
+        Index index = createIndex("testGetTypoTolerance");
+        Settings initialSettings = index.getSettings();
+        TypoTolerance initialTypoTolerance = index.getTypoToleranceSettings();
 
-        index.waitForTask(index.updateRankingRuleSettings(newRankingRules).getUid());
-        String[] newRankingRule = index.getRankingRuleSettings();
+        assertEquals(initialSettings.getTypoTolerance().getDisableOnWords().length, 0);
+        assertEquals(initialTypoTolerance.getDisableOnWords().length, 0);
+        assertEquals(initialSettings.getTypoTolerance().getDisableOnAttributes().length, 0);
+        assertEquals(initialTypoTolerance.getDisableOnAttributes().length, 0);
+        assertEquals(
+                initialSettings.getTypoTolerance().isEnabled(), initialTypoTolerance.isEnabled());
+        assertNotNull(initialTypoTolerance.getMinWordSizeForTypos().containsKey("oneTypo"));
+        assertNotNull(initialTypoTolerance.getMinWordSizeForTypos().get("oneTypo"));
+        assertNotNull(initialTypoTolerance.getMinWordSizeForTypos().containsKey("twoTypos"));
+        assertNotNull(initialTypoTolerance.getMinWordSizeForTypos().get("twoTypos"));
+    }
+
+    @Test
+    @DisplayName("Test update typo tolerance settings")
+    public void testUpdateTypoTolerance() throws Exception {
```

Comment:
Can you change this test (or add another) with the update of just a couple of fields like only `setDisableOnAttributes` and `minWordSizeTypos`? Just to ensure all the other fields remain the same (because the update is partial)

Reply:
I already check this field, you mean check it before the update?

Reply:
The assertions phase is great, I mean the exercise part is my point.

Your test is perfect as well, I just would like to have another test that is more "real-life", which means in real life the users may not update all the fields at once like the test is doing, they maybe just change one or two, and my suggestion is to have test with this kind of case.

I don't know if it was clear enough, but let me know!

Reply:
btw @alallema this is the last comment, after this I'll approve it 👯